src/choice_map.jl

```diff
@@ -953,3 +953,24 @@ nested_view(c::ChoiceMap) = ChoiceMapNestedView(c)
 # aux data together.
 
 export nested_view
+
+#############################################
+# filtering choice maps based on selections #
+#############################################
+
+function get_selected(
```

Comment:
This is a new function, right?  Do we want to add it to the public API?  (If so let's add a brief docstring)

Reply:
Good point.

Reply:
I added it to the public API for choice maps, and also added a function `complement()` to the public API for selections.